From ed203f4d5269b482d7ef3b91e8932f9570262c68 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Wed, 10 Jul 2024 15:17:36 +0200
Subject: [PATCH 656/954] feat(st-clock): rename stm32mp2_a35_ss_on_hsi

Rename the function stm32mp2_a35_ss_on_hsi to stm32mp2_a35_ss_on_bypass
to reflect its behavior; switch the CA35 clock on the bypass clock aka
ck_cpu1_ext2f = flexgen63.

The function switch to HSI only if FLEXGEN63 MUX is defined to HSI source.
It is assumed on cold boot / reset but it is not the case for STOP2 exit.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 2df45a8161319849994c7f3d53ce7303932af5d5 feat(st-clock): rename stm32mp2_a35_ss_on_hsi]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Id09cdf00fdae6a30a2db454fb2e7c33660cb1356
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/392692
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/st/clk/clk-stm32mp2.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/st/clk/clk-stm32mp2.c b/drivers/st/clk/clk-stm32mp2.c
index 4f3fdef5de..981ed830ac 100644
--- a/drivers/st/clk/clk-stm32mp2.c
+++ b/drivers/st/clk/clk-stm32mp2.c
@@ -1498,7 +1498,8 @@ enum clksrc_id {
 	CLKSRC_NB
 };
 
-static void stm32mp2_a35_ss_on_hsi(void)
+/* switch CA35 clock on bypass clock aka ck_cpu1_ext2f = flexgen63 */
+static void stm32mp2_a35_ss_on_bypass(void)
 {
 	uintptr_t a35_ss_address = A35SSC_BASE;
 	uintptr_t chgclkreq_reg = a35_ss_address + A35_SS_CHGCLKREQ;
@@ -1748,7 +1749,7 @@ static int _clk_stm32_pll1_init(struct stm32_clk_priv *priv, int pll_idx,
 	    a configuration on the fly.
 	 */
 
-	stm32mp2_a35_ss_on_hsi();
+	stm32mp2_a35_ss_on_bypass();
 
 	ret = stm32_clk_configure_mux(priv, pll_conf->src);
 	if (ret != 0) {
@@ -2081,7 +2082,8 @@ static void stm32_enable_oscillator_lse(struct stm32_clk_priv *priv)
 
 static int stm32mp2_clk_switch_to_hsi(struct stm32_clk_priv *priv)
 {
-	stm32mp2_a35_ss_on_hsi();
+	/* on reset, bypass use HSI for CA35 clock (FLEXGEN63 = ck_cpu1_ext2f) */
+	stm32mp2_a35_ss_on_bypass();
 	stm32mp2_clk_muxsel_on_hsi(priv);
 #if !STM32MP_M33_TDCID
 	stm32mp2_clk_xbar_on_hsi(priv);
@@ -2586,7 +2588,7 @@ int stm32mp2_pll1_disable(void)
 	saved_pll_freq1_reg = mmio_read_32(pll_freq1_reg);
 	saved_pll_freq2_reg = mmio_read_32(pll_freq2_reg);
 
-	stm32mp2_a35_ss_on_hsi();
+	stm32mp2_a35_ss_on_bypass();
 
 	mmio_clrbits_32(pll_enable_reg, A35_SS_PLL_ENABLE_PD);
 
-- 
2.53.0

